mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -165,6 +168,32 @@ def __init__(self, name=None, bus=None, use_settings=True):
         # Skill Public API
         self.public_api = {}
 
+        if bus:
+            self._startup(bus)
+
+    def _startup(self, bus):
+        """Startup the skill.
+
+        This connects the skill to the messagebus, loads vocabularies and
+        data files and in the end calls the skill creator's "intialize" code.
+
+        Arguments:
+            bus: Mycroft Messagebus connection object.
+        """
+        self.bind(bus)
+        try:
+            self.load_data_files()
+            # Set up intent handlers
+            self._register_decorated()
+            self.register_resting_screen()
+            self.initialize()
+        except Exception as e:
+            # If an exception occurs, make sure to clean up the skill
+            self.default_shutdown()
+            log_msg = 'Skill initialization failed with {}'
+            LOG.exception(log_msg.format(repr(e)))
+            raise e
```

Comment:
you don't need to pass the exception to the raise statement

Reply:
Will fix, this part was moved from the old skill loading without any changes